# Исследование объявлений о продаже квартир

### 0.1 Шаг 1. Откройте файл с данными и изучите общую информацию

In [1]:
import pandas as pd
from geopy.distance import geodesic as gd
from geopy.geocoders import Nominatim as nm
import time
data = pd.read_csv('real_estate_data.csv', sep='\t')

print('Информация о датафрейме:')
data.shape

Информация о датафрейме:


(23699, 22)

In [2]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 23699 entries, 0 to 23698
Data columns (total 22 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   total_images          23699 non-null  int64  
 1   last_price            23699 non-null  float64
 2   total_area            23699 non-null  float64
 3   first_day_exposition  23699 non-null  object 
 4   rooms                 23699 non-null  int64  
 5   ceiling_height        14504 non-null  float64
 6   floors_total          23613 non-null  float64
 7   living_area           21796 non-null  float64
 8   floor                 23699 non-null  int64  
 9   is_apartment          2775 non-null   object 
 10  studio                23699 non-null  bool   
 11  open_plan             23699 non-null  bool   
 12  kitchen_area          21421 non-null  float64
 13  balcony               12180 non-null  float64
 14  locality_name         23650 non-null  object 
 15  airports_nearest   

In [3]:
display(data.head(30))

total_images  last_price  total_area first_day_exposition  rooms  \
0             20  13000000.0      108.00  2019-03-07T00:00:00      3   
1              7   3350000.0       40.40  2018-12-04T00:00:00      1   
2             10   5196000.0       56.00  2015-08-20T00:00:00      2   
3              0  64900000.0      159.00  2015-07-24T00:00:00      3   
4              2  10000000.0      100.00  2018-06-19T00:00:00      2   
5             10   2890000.0       30.40  2018-09-10T00:00:00      1   
6              6   3700000.0       37.30  2017-11-02T00:00:00      1   
7              5   7915000.0       71.60  2019-04-18T00:00:00      2   
8             20   2900000.0       33.16  2018-05-23T00:00:00      1   
9             18   5400000.0       61.00  2017-02-26T00:00:00      3   
10             5   5050000.0       39.60  2017-11-16T00:00:00      1   
11             9   3300000.0       44.00  2018-08-27T00:00:00      2   
12            10   3890000.0       54.00  2016-06-30T00:00:00      2   
13            20   3550000.0       42.80  2017-07-01T00:00:00      2   
14             1   4400000.0       36.00  2016-06-23T00:00:00      1   
15            16   4650000.0       39.00  2017-11-18T00:00:00      1   
16            11   6700000.0       82.00  2017-11-23T00:00:00      3   
17             6   4180000.0       36.00  2016-09-09T00:00:00      1   
18             8   3250000.0       31.00  2017-01-27T00:00:00      1   
19            16  14200000.0      121.00  2019-01-09T00:00:00      3   
20            12   6120000.0       80.00  2017-09-28T00:00:00      3   
21            13   3200000.0       31.60  2018-03-14T00:00:00      1   
22            20   5000000.0       58.00  2017-04-24T00:00:00      2   
23            11   2950000.0       32.00  2016-10-29T00:00:00      1   
24             8   6500000.0       97.20  2015-10-31T00:00:00      2   
25             3   6800000.0       76.00  2015-10-01T00:00:00      2   
26             6   4050000.0       60.00  2017-04-28T00:00:00      4   
27            20   7100000.0       70.00  2017-05-12T00:00:00      3   
28             8   4170000.0       44.00  2017-12-13T00:00:00      1   
29             9   8600000.0      100.00  2016-04-09T00:00:00      3   

    ceiling_height  floors_total  living_area  floor is_apartment  ...  \
0             2.70          16.0        51.00      8          NaN  ...   
1              NaN          11.0        18.60      1          NaN  ...   
2              NaN           5.0        34.30      4          NaN  ...   
3              NaN          14.0          NaN      9          NaN  ...   
4             3.03          14.0        32.00     13          NaN  ...   
5              NaN          12.0        14.40      5          NaN  ...   
6              NaN          26.0        10.60      6          NaN  ...   
7              NaN          24.0          NaN     22          NaN  ...   
8              NaN          27.0        15.43     26          NaN  ...   
9             2.50           9.0        43.60      7          NaN  ...   
10            2.67          12.0        20.30      3          NaN  ...   
11             NaN           5.0        31.00      4        False  ...   
12             NaN           5.0        30.00      5          NaN  ...   
13            2.56           5.0        27.00      5          NaN  ...   
14             NaN           6.0        17.00      1          NaN  ...   
15             NaN          14.0        20.50      5          NaN  ...   
16            3.05           5.0        55.60      1          NaN  ...   
17             NaN          17.0        16.50      7          NaN  ...   
18            2.50           5.0        19.40      2          NaN  ...   
19            2.75          16.0        76.00      8          NaN  ...   
20            2.70          27.0        48.00     11          NaN  ...   
21             NaN           5.0        16.90      2          NaN  ...   
22            2.75          25.0        30.00     15          NaN  ..

In [4]:
data.columns

Index(['total_images', 'last_price', 'total_area', 'first_day_exposition',
       'rooms', 'ceiling_height', 'floors_total', 'living_area', 'floor',
       'is_apartment', 'studio', 'open_plan', 'kitchen_area', 'balcony',
       'locality_name', 'airports_nearest', 'cityCenters_nearest',
       'parks_around3000', 'parks_nearest', 'ponds_around3000',
       'ponds_nearest', 'days_exposition'],
      dtype='object')

In [5]:
data.columns[int(len(data.columns) / 2)]

'open_plan'

<div class="alert alert-info">
<b>Комментарий студента:</b>
<br>
Судя на третьей строчке проектной работы по установлению параметров и проведению предобработки и исследовательского анализа данных с архивными объявлениями о продаже квартир в г. Санкт-Петербург: мы обнаруживаем, что вместо представления данных в столбцах: 'studio' (наличие квартиры-студии) и 'open_plan' (наличие свободной планировки) обнаруживание пропуск данных в виде многоточия ('...') и неполную картину над параметрами всех объявлений о продаже квартир в г. Санкт-Петербург<br>
<br>
Поэтому разделим датафрейм data на две части с одинаковым количество строк, где первая часть заканчивается, а вторая часть начинается с одного и того же столбца, а именно с 'open_plan' (из пятой строки), чтобы не запутаться.<br>
<br>
Затем создадим два списка для разделенения всех названий столбцов в данной таблице (first_part и second_part) и цикл, по которому итерация помещает одно названия столбца в один список, а другое название - в другой список и выводим по частям два разделённых датафрейма: first_part_data и second_part_data для полного прочтения всех значений датафрейма data.<br>
<br>
Ещё в второй и четвёртой строчке данной проектной работы, обнаруживаем неэтическое для программистов название столбца: 'cityCenters_nearest', поэтому сразу меняем это название на 'city_сenters_nearest'.<br>
</div>

In [6]:
#data[['cityCenters_nearest']] = data[['city_centers_nearest']]

In [7]:
first_part = []
second_part = []

for i in range(len(data.columns)):
    if(i <= len(data.columns) / 2 - 1):
        first_part.append(data.columns[i])
    elif(i == len(data.columns) / 2):
        first_part.append(data.columns[i])
        second_part.append(data.columns[i])
    else:
        second_part.append(data.columns[i])

In [8]:
first_part_data = data.loc[:, first_part]
display(first_part_data.head(30))

total_images  last_price  total_area first_day_exposition  rooms  \
0             20  13000000.0      108.00  2019-03-07T00:00:00      3   
1              7   3350000.0       40.40  2018-12-04T00:00:00      1   
2             10   5196000.0       56.00  2015-08-20T00:00:00      2   
3              0  64900000.0      159.00  2015-07-24T00:00:00      3   
4              2  10000000.0      100.00  2018-06-19T00:00:00      2   
5             10   2890000.0       30.40  2018-09-10T00:00:00      1   
6              6   3700000.0       37.30  2017-11-02T00:00:00      1   
7              5   7915000.0       71.60  2019-04-18T00:00:00      2   
8             20   2900000.0       33.16  2018-05-23T00:00:00      1   
9             18   5400000.0       61.00  2017-02-26T00:00:00      3   
10             5   5050000.0       39.60  2017-11-16T00:00:00      1   
11             9   3300000.0       44.00  2018-08-27T00:00:00      2   
12            10   3890000.0       54.00  2016-06-30T00:00:00      2   
13            20   3550000.0       42.80  2017-07-01T00:00:00      2   
14             1   4400000.0       36.00  2016-06-23T00:00:00      1   
15            16   4650000.0       39.00  2017-11-18T00:00:00      1   
16            11   6700000.0       82.00  2017-11-23T00:00:00      3   
17             6   4180000.0       36.00  2016-09-09T00:00:00      1   
18             8   3250000.0       31.00  2017-01-27T00:00:00      1   
19            16  14200000.0      121.00  2019-01-09T00:00:00      3   
20            12   6120000.0       80.00  2017-09-28T00:00:00      3   
21            13   3200000.0       31.60  2018-03-14T00:00:00      1   
22            20   5000000.0       58.00  2017-04-24T00:00:00      2   
23            11   2950000.0       32.00  2016-10-29T00:00:00      1   
24             8   6500000.0       97.20  2015-10-31T00:00:00      2   
25             3   6800000.0       76.00  2015-10-01T00:00:00      2   
26             6   4050000.0       60.00  2017-04-28T00:00:00      4   
27            20   7100000.0       70.00  2017-05-12T00:00:00      3   
28             8   4170000.0       44.00  2017-12-13T00:00:00      1   
29             9   8600000.0      100.00  2016-04-09T00:00:00      3   

    ceiling_height  floors_total  living_area  floor is_apartment  studio  \
0             2.70          16.0        51.00      8          NaN   False   
1              NaN          11.0        18.60      1          NaN   False   
2              NaN           5.0        34.30      4          NaN   False   
3              NaN          14.0          NaN      9          NaN   False   
4             3.03          14.0        32.00     13          NaN   False   
5              NaN          12.0        14.40      5          NaN   False   
6              NaN          26.0        10.60      6          NaN   False   
7              NaN          24.0          NaN     22          NaN   False   
8              NaN          27.0        15.43     26          NaN   False   
9             2.50           9.0        43.60      7          NaN   False   
10            2.67          12.0        20.30      3          NaN   False   
11             NaN           5.0        31.00      4        False   False   
12             NaN           5.0        30.00      5          NaN   False   
13            2.56           5.0        27.00      5          NaN   False   
14             NaN           6.0        17.00      1          NaN   False   
15             NaN          14.0        20.50      5          NaN   False   
16            3.05           5.0        55.60      1          NaN   False   
17             NaN          17.0        16.50      7          NaN   False   
18            2.50           5.0        19.40      2          NaN   False   
19            2.75          16.0        76.00      8          NaN   False   
20            2.70          27.0        48.00     11          NaN   False   
21             NaN           5.0        16.90      2          NaN   False   


In [9]:
second_part_data = data.loc[:, second_part]
display(second_part_data.head(30))

open_plan  kitchen_area  balcony              locality_name  \
0       False         25.00      NaN            Санкт-Петербург   
1       False         11.00      2.0             посёлок Шушары   
2       False          8.30      0.0            Санкт-Петербург   
3       False           NaN      0.0            Санкт-Петербург   
4       False         41.00      NaN            Санкт-Петербург   
5       False          9.10      NaN  городской посёлок Янино-1   
6       False         14.40      1.0          посёлок Парголово   
7       False         18.90      2.0            Санкт-Петербург   
8       False          8.81      NaN             посёлок Мурино   
9       False          6.50      2.0            Санкт-Петербург   
10      False          8.50      NaN            Санкт-Петербург   
11      False          6.00      1.0                  Ломоносов   
12      False          9.00      0.0                  Сертолово   
13      False          5.20      1.0                   Петергоф   
14      False          8.00      0.0                     Пушкин   
15      False          7.60      1.0            Санкт-Петербург   
16      False          9.00      NaN            Санкт-Петербург   
17      False         11.00      1.0            Санкт-Петербург   
18      False          5.60      1.0            Санкт-Петербург   
19      False         12.00      NaN            Санкт-Петербург   
20      False         12.00      2.0          посёлок Парголово   
21      False          5.70      1.0            Санкт-Петербург   
22      False         11.00      2.0            деревня Кудрово   
23      False          6.10      NaN            Санкт-Петербург   
24      False         19.60      0.0            Санкт-Петербург   
25      False         15.00      2.0            Санкт-Петербург   
26      False          7.00      NaN            Санкт-Петербург   
27      False          9.20      0.0            Санкт-Петербург   
28      False         11.70      1.0                     Пушкин   
29      False         11.00      0.0            Санкт-Петербург   

    airports_nearest  cityCenters_nearest  parks_around3000  parks_nearest  \
0            18863.0              16028.0               1.0          482.0   
1            12817.0              18603.0               0.0            NaN   
2            21741.0              13933.0               1.0           90.0   
3            28098.0               6800.0               2.0           84.0   
4            31856.0               8098.0               2.0          112.0   
5                NaN                  NaN               NaN            NaN   
6            52996.0              19143.0               0.0            NaN   
7            23982.0              11634.0               0.0            NaN   
8                NaN                  NaN               NaN            NaN   
9            50898.0              15008.0               0.0            NaN   
10           38357.0              13878.0               1.0          310.0   
11           48252.0              51677.0               0.0            NaN   
12               NaN                  NaN               NaN            NaN   
13           37868.0              33058.0               1.0          294.0   
14           20782.0              30759.0               0.0            NaN   
15           12900.0              14259.0               1.0          590.0   
16           22108.0              10698.0               3.0          420.0   
17           33564.0              14616.0               0.0            NaN   
18           44060.0              10842.0               1.0          759.0   
19           38900.0              12843.0               0.0            NaN   
20           53134.0              19311.0               0.0            NaN   
21           36064.0               9722.0               1.0          248.0   
22               NaN                  NaN               NaN            NaN   
23           15414.0              14211.0             

 ### 0.2 Шаг 2.1 Предобработка данных

<div class="alert alert-info">
<b>Комментарий студента:</b>
<br>
Судя на третьей строчке проектной работы по установлению параметров и проведению предобработки и исследовательского анализа данных с архивными объявлениями о продаже квартир в г. Санкт-Петербург: мы обнаруживаем, что вместо представления данных в столбцах: 'studio' (наличие квартиры-студии) и 'open_plan' (наличие свободной планировки) обнаруживание пропуск данных в виде многоточия ('...') и неполную картину над параметрами всех объявлений о продаже квартир в г. Санкт-Петербург<br>
<br>
Поэтому разделим датафрейм data на две части с одинаковым количество строк, где первая часть заканчивается, а вторая часть начинается с одного и того же столбца, а именно с 'open_plan' (из пятой строки).<br>
<br>
Затем создадим два списка для разделенения всех названий столбцов в данной таблице (first_part и second_part) и цикл, по которому итерация помещает одно названия столбца в один список, а другое название - в другой список и выводим по частям два разделённых датафрейма: first_part_data и second_part_data для полного прочтения всех значений датафрейма data.<br>
<br>
Ещё в второй и четвёртой строчке данной проектной работы, обнаруживаем неэтическое для программистов название столбца: 'cityCenters_nearest', поэтому сразу меняем это название на 'city_сenters_nearest'.<br>
</div>

In [10]:
data.isna().sum()

total_images                0
last_price                  0
total_area                  0
first_day_exposition        0
rooms                       0
ceiling_height           9195
floors_total               86
living_area              1903
floor                       0
is_apartment            20924
studio                      0
open_plan                   0
kitchen_area             2278
balcony                 11519
locality_name              49
airports_nearest         5542
cityCenters_nearest      5519
parks_around3000         5518
parks_nearest           15620
ponds_around3000         5518
ponds_nearest           14589
days_exposition          3181
dtype: int64

In [11]:
null_columns = []

for value in data.columns: 
    if data[value].count() != data.shape[0]:
        null_columns.append(value)
        
null_columns

['ceiling_height',
 'floors_total',
 'living_area',
 'is_apartment',
 'kitchen_area',
 'balcony',
 'locality_name',
 'airports_nearest',
 'cityCenters_nearest',
 'parks_around3000',
 'parks_nearest',
 'ponds_around3000',
 'ponds_nearest',
 'days_exposition']

In [12]:
for item in null_columns:
    print(f"\n Строки датафрейма, где у столбца '{item}' обнаружены пропущенные значения:")
    display(first_part_data[data[item].isna()].head())
    display(second_part_data[data[item].isna()].head())


 Строки датафрейма, где у столбца 'ceiling_height' обнаружены пропущенные значения:


total_images  last_price  total_area first_day_exposition  rooms  \
1             7   3350000.0        40.4  2018-12-04T00:00:00      1   
2            10   5196000.0        56.0  2015-08-20T00:00:00      2   
3             0  64900000.0       159.0  2015-07-24T00:00:00      3   
5            10   2890000.0        30.4  2018-09-10T00:00:00      1   
6             6   3700000.0        37.3  2017-11-02T00:00:00      1   

   ceiling_height  floors_total  living_area  floor is_apartment  studio  \
1             NaN          11.0         18.6      1          NaN   False   
2             NaN           5.0         34.3      4          NaN   False   
3             NaN          14.0          NaN      9          NaN   False   
5             NaN          12.0         14.4      5          NaN   False   
6             NaN          26.0         10.6      6          NaN   False   

   open_plan  
1      False  
2      False  
3      False  
5      False  
6      False

open_plan  kitchen_area  balcony              locality_name  \
1      False          11.0      2.0             посёлок Шушары   
2      False           8.3      0.0            Санкт-Петербург   
3      False           NaN      0.0            Санкт-Петербург   
5      False           9.1      NaN  городской посёлок Янино-1   
6      False          14.4      1.0          посёлок Парголово   

   airports_nearest  cityCenters_nearest  parks_around3000  parks_nearest  \
1           12817.0              18603.0               0.0            NaN   
2           21741.0              13933.0               1.0           90.0   
3           28098.0               6800.0               2.0           84.0   
5               NaN                  NaN               NaN            NaN   
6           52996.0              19143.0               0.0            NaN   

   ponds_around3000  ponds_nearest  days_exposition  
1               0.0            NaN             81.0  
2               2.0          574.0            558.0  
3               3.0          234.0            424.0  
5               NaN            NaN             55.0  
6               0.0            NaN            155.0


 Строки датафрейма, где у столбца 'floors_total' обнаружены пропущенные значения:


total_images  last_price  total_area first_day_exposition  rooms  \
186             12  11640000.0        65.2  2018-10-02T00:00:00      2   
237              4   2438033.0        28.1  2016-11-23T00:00:00      1   
457              4   9788348.0        70.8  2015-08-01T00:00:00      2   
671              4   6051191.0        93.6  2017-04-06T00:00:00      3   
1757             5   3600000.0        39.0  2017-04-22T00:00:00      1   

      ceiling_height  floors_total  living_area  floor is_apartment  studio  \
186              NaN           NaN        30.80      4          NaN   False   
237              NaN           NaN        20.75      1          NaN   False   
457              NaN           NaN        38.40     12          NaN   False   
671              NaN           NaN        47.10      8          NaN   False   
1757             NaN           NaN          NaN      9          NaN   False   

      open_plan  
186       False  
237       False  
457       False  
671       False  
1757      False

open_plan  kitchen_area  balcony    locality_name  airports_nearest  \
186       False         12.00      NaN  Санкт-Петербург           39197.0   
237       False           NaN      0.0  Санкт-Петербург           22041.0   
457       False         10.63      0.0  Санкт-Петербург           37364.0   
671       False         16.80      0.0  Санкт-Петербург           22041.0   
1757      False           NaN      0.0  Санкт-Петербург           22735.0   

      cityCenters_nearest  parks_around3000  parks_nearest  ponds_around3000  \
186               12373.0               1.0          123.0               0.0   
237               17369.0               0.0            NaN               1.0   
457                8322.0               2.0          309.0               2.0   
671               17369.0               0.0            NaN               1.0   
1757              11618.0               1.0          835.0               1.0   

      ponds_nearest  days_exposition  
186             NaN             49.0  
237           374.0            251.0  
457           706.0            727.0  
671           374.0            123.0  
1757          652.0             77.0


 Строки датафрейма, где у столбца 'living_area' обнаружены пропущенные значения:


total_images  last_price  total_area first_day_exposition  rooms  \
3              0  64900000.0       159.0  2015-07-24T00:00:00      3   
7              5   7915000.0        71.6  2019-04-18T00:00:00      2   
30            12   2200000.0        32.8  2018-02-19T00:00:00      1   
37            10   1990000.0        45.8  2017-10-28T00:00:00      2   
44            13   5350000.0        40.0  2018-11-18T00:00:00      1   

    ceiling_height  floors_total  living_area  floor is_apartment  studio  \
3              NaN          14.0          NaN      9          NaN   False   
7              NaN          24.0          NaN     22          NaN   False   
30             NaN           9.0          NaN      2          NaN   False   
37             2.5           5.0          NaN      1          NaN   False   
44             NaN          22.0          NaN      3          NaN   False   

    open_plan  
3       False  
7       False  
30      False  
37      False  
44      False

open_plan  kitchen_area  balcony                        locality_name  \
3       False           NaN      0.0                      Санкт-Петербург   
7       False          18.9      2.0                      Санкт-Петербург   
30      False           NaN      NaN                             Коммунар   
37      False           NaN      NaN  поселок городского типа Красный Бор   
44      False           NaN      1.0                      Санкт-Петербург   

    airports_nearest  cityCenters_nearest  parks_around3000  parks_nearest  \
3            28098.0               6800.0               2.0           84.0   
7            23982.0              11634.0               0.0            NaN   
30               NaN                  NaN               NaN            NaN   
37               NaN                  NaN               NaN            NaN   
44           30471.0              11603.0               1.0          620.0   

    ponds_around3000  ponds_nearest  days_exposition  
3                3.0          234.0            424.0  
7                0.0            NaN              NaN  
30               NaN            NaN             63.0  
37               NaN            NaN            196.0  
44               1.0         1152.0              NaN


 Строки датафрейма, где у столбца 'is_apartment' обнаружены пропущенные значения:


total_images  last_price  total_area first_day_exposition  rooms  \
0            20  13000000.0       108.0  2019-03-07T00:00:00      3   
1             7   3350000.0        40.4  2018-12-04T00:00:00      1   
2            10   5196000.0        56.0  2015-08-20T00:00:00      2   
3             0  64900000.0       159.0  2015-07-24T00:00:00      3   
4             2  10000000.0       100.0  2018-06-19T00:00:00      2   

   ceiling_height  floors_total  living_area  floor is_apartment  studio  \
0            2.70          16.0         51.0      8          NaN   False   
1             NaN          11.0         18.6      1          NaN   False   
2             NaN           5.0         34.3      4          NaN   False   
3             NaN          14.0          NaN      9          NaN   False   
4            3.03          14.0         32.0     13          NaN   False   

   open_plan  
0      False  
1      False  
2      False  
3      False  
4      False

open_plan  kitchen_area  balcony    locality_name  airports_nearest  \
0      False          25.0      NaN  Санкт-Петербург           18863.0   
1      False          11.0      2.0   посёлок Шушары           12817.0   
2      False           8.3      0.0  Санкт-Петербург           21741.0   
3      False           NaN      0.0  Санкт-Петербург           28098.0   
4      False          41.0      NaN  Санкт-Петербург           31856.0   

   cityCenters_nearest  parks_around3000  parks_nearest  ponds_around3000  \
0              16028.0               1.0          482.0               2.0   
1              18603.0               0.0            NaN               0.0   
2              13933.0               1.0           90.0               2.0   
3               6800.0               2.0           84.0               3.0   
4               8098.0               2.0          112.0               1.0   

   ponds_nearest  days_exposition  
0          755.0              NaN  
1            NaN             81.0  
2          574.0            558.0  
3          234.0            424.0  
4           48.0            121.0


 Строки датафрейма, где у столбца 'kitchen_area' обнаружены пропущенные значения:


total_images  last_price  total_area first_day_exposition  rooms  \
3              0  64900000.0       159.0  2015-07-24T00:00:00      3   
30            12   2200000.0        32.8  2018-02-19T00:00:00      1   
37            10   1990000.0        45.8  2017-10-28T00:00:00      2   
44            13   5350000.0        40.0  2018-11-18T00:00:00      1   
59            15   6300000.0        46.5  2017-03-06T00:00:00      1   

    ceiling_height  floors_total  living_area  floor is_apartment  studio  \
3              NaN          14.0          NaN      9          NaN   False   
30             NaN           9.0          NaN      2          NaN   False   
37             2.5           5.0          NaN      1          NaN   False   
44             NaN          22.0          NaN      3          NaN   False   
59             NaN          13.0          NaN     13          NaN   False   

    open_plan  
3       False  
30      False  
37      False  
44      False  
59      False

open_plan  kitchen_area  balcony                        locality_name  \
3       False           NaN      0.0                      Санкт-Петербург   
30      False           NaN      NaN                             Коммунар   
37      False           NaN      NaN  поселок городского типа Красный Бор   
44      False           NaN      1.0                      Санкт-Петербург   
59      False           NaN      NaN                      Санкт-Петербург   

    airports_nearest  cityCenters_nearest  parks_around3000  parks_nearest  \
3            28098.0               6800.0               2.0           84.0   
30               NaN                  NaN               NaN            NaN   
37               NaN                  NaN               NaN            NaN   
44           30471.0              11603.0               1.0          620.0   
59           11125.0              14501.0               0.0            NaN   

    ponds_around3000  ponds_nearest  days_exposition  
3                3.0          234.0            424.0  
30               NaN            NaN             63.0  
37               NaN            NaN            196.0  
44               1.0         1152.0              NaN  
59               0.0            NaN            336.0


 Строки датафрейма, где у столбца 'balcony' обнаружены пропущенные значения:


total_images  last_price  total_area first_day_exposition  rooms  \
0             20  13000000.0      108.00  2019-03-07T00:00:00      3   
4              2  10000000.0      100.00  2018-06-19T00:00:00      2   
5             10   2890000.0       30.40  2018-09-10T00:00:00      1   
8             20   2900000.0       33.16  2018-05-23T00:00:00      1   
10             5   5050000.0       39.60  2017-11-16T00:00:00      1   

    ceiling_height  floors_total  living_area  floor is_apartment  studio  \
0             2.70          16.0        51.00      8          NaN   False   
4             3.03          14.0        32.00     13          NaN   False   
5              NaN          12.0        14.40      5          NaN   False   
8              NaN          27.0        15.43     26          NaN   False   
10            2.67          12.0        20.30      3          NaN   False   

    open_plan  
0       False  
4       False  
5       False  
8       False  
10      False

open_plan  kitchen_area  balcony              locality_name  \
0       False         25.00      NaN            Санкт-Петербург   
4       False         41.00      NaN            Санкт-Петербург   
5       False          9.10      NaN  городской посёлок Янино-1   
8       False          8.81      NaN             посёлок Мурино   
10      False          8.50      NaN            Санкт-Петербург   

    airports_nearest  cityCenters_nearest  parks_around3000  parks_nearest  \
0            18863.0              16028.0               1.0          482.0   
4            31856.0               8098.0               2.0          112.0   
5                NaN                  NaN               NaN            NaN   
8                NaN                  NaN               NaN            NaN   
10           38357.0              13878.0               1.0          310.0   

    ponds_around3000  ponds_nearest  days_exposition  
0                2.0          755.0              NaN  
4                1.0           48.0            121.0  
5                NaN            NaN             55.0  
8                NaN            NaN            189.0  
10               2.0          553.0            137.0


 Строки датафрейма, где у столбца 'locality_name' обнаружены пропущенные значения:


total_images  last_price  total_area first_day_exposition  rooms  \
1097             3   8600000.0        81.7  2016-04-15T00:00:00      3   
2033             6   5398000.0        80.0  2017-05-30T00:00:00      3   
2603            20   3351765.0        42.7  2015-09-20T00:00:00      1   
2632             2   5130593.0        62.4  2015-10-11T00:00:00      2   
3574            10   4200000.0        46.5  2016-05-28T00:00:00      2   

      ceiling_height  floors_total  living_area  floor is_apartment  studio  \
1097            3.55           5.0         50.8      2          NaN   False   
2033             NaN           4.0         42.6      2          NaN   False   
2603             NaN          24.0         15.6      3          NaN   False   
2632             NaN          24.0         33.1     21          NaN   False   
3574             NaN           5.0         30.8      5          NaN   False   

      open_plan  
1097      False  
2033      False  
2603      False  
2632      False  
3574      False

open_plan  kitchen_area  balcony locality_name  airports_nearest  \
1097      False           8.8      0.0           NaN           23478.0   
2033      False          18.6      0.0           NaN               NaN   
2603      False          10.7      0.0           NaN           22041.0   
2632      False           8.2      0.0           NaN           22041.0   
3574      False           6.5      0.0           NaN           27419.0   

      cityCenters_nearest  parks_around3000  parks_nearest  ponds_around3000  \
1097               4258.0               0.0            NaN               0.0   
2033                  NaN               NaN            NaN               NaN   
2603              17369.0               0.0            NaN               1.0   
2632              17369.0               0.0            NaN               1.0   
3574               8127.0               0.0            NaN               1.0   

      ponds_nearest  days_exposition  
1097            NaN            147.0  
2033            NaN             34.0  
2603          374.0            276.0  
2632          374.0            256.0  
3574          603.0             45.0


 Строки датафрейма, где у столбца 'airports_nearest' обнаружены пропущенные значения:


total_images  last_price  total_area first_day_exposition  rooms  \
5             10   2890000.0       30.40  2018-09-10T00:00:00      1   
8             20   2900000.0       33.16  2018-05-23T00:00:00      1   
12            10   3890000.0       54.00  2016-06-30T00:00:00      2   
22            20   5000000.0       58.00  2017-04-24T00:00:00      2   
30            12   2200000.0       32.80  2018-02-19T00:00:00      1   

    ceiling_height  floors_total  living_area  floor is_apartment  studio  \
5              NaN          12.0        14.40      5          NaN   False   
8              NaN          27.0        15.43     26          NaN   False   
12             NaN           5.0        30.00      5          NaN   False   
22            2.75          25.0        30.00     15          NaN   False   
30             NaN           9.0          NaN      2          NaN   False   

    open_plan  
5       False  
8       False  
12      False  
22      False  
30      False

open_plan  kitchen_area  balcony              locality_name  \
5       False          9.10      NaN  городской посёлок Янино-1   
8       False          8.81      NaN             посёлок Мурино   
12      False          9.00      0.0                  Сертолово   
22      False         11.00      2.0            деревня Кудрово   
30      False           NaN      NaN                   Коммунар   

    airports_nearest  cityCenters_nearest  parks_around3000  parks_nearest  \
5                NaN                  NaN               NaN            NaN   
8                NaN                  NaN               NaN            NaN   
12               NaN                  NaN               NaN            NaN   
22               NaN                  NaN               NaN            NaN   
30               NaN                  NaN               NaN            NaN   

    ponds_around3000  ponds_nearest  days_exposition  
5                NaN            NaN             55.0  
8                NaN            NaN            189.0  
12               NaN            NaN             90.0  
22               NaN            NaN             60.0  
30               NaN            NaN             63.0


 Строки датафрейма, где у столбца 'cityCenters_nearest' обнаружены пропущенные значения:


total_images  last_price  total_area first_day_exposition  rooms  \
5             10   2890000.0       30.40  2018-09-10T00:00:00      1   
8             20   2900000.0       33.16  2018-05-23T00:00:00      1   
12            10   3890000.0       54.00  2016-06-30T00:00:00      2   
22            20   5000000.0       58.00  2017-04-24T00:00:00      2   
30            12   2200000.0       32.80  2018-02-19T00:00:00      1   

    ceiling_height  floors_total  living_area  floor is_apartment  studio  \
5              NaN          12.0        14.40      5          NaN   False   
8              NaN          27.0        15.43     26          NaN   False   
12             NaN           5.0        30.00      5          NaN   False   
22            2.75          25.0        30.00     15          NaN   False   
30             NaN           9.0          NaN      2          NaN   False   

    open_plan  
5       False  
8       False  
12      False  
22      False  
30      False

open_plan  kitchen_area  balcony              locality_name  \
5       False          9.10      NaN  городской посёлок Янино-1   
8       False          8.81      NaN             посёлок Мурино   
12      False          9.00      0.0                  Сертолово   
22      False         11.00      2.0            деревня Кудрово   
30      False           NaN      NaN                   Коммунар   

    airports_nearest  cityCenters_nearest  parks_around3000  parks_nearest  \
5                NaN                  NaN               NaN            NaN   
8                NaN                  NaN               NaN            NaN   
12               NaN                  NaN               NaN            NaN   
22               NaN                  NaN               NaN            NaN   
30               NaN                  NaN               NaN            NaN   

    ponds_around3000  ponds_nearest  days_exposition  
5                NaN            NaN             55.0  
8                NaN            NaN            189.0  
12               NaN            NaN             90.0  
22               NaN            NaN             60.0  
30               NaN            NaN             63.0


 Строки датафрейма, где у столбца 'parks_around3000' обнаружены пропущенные значения:


total_images  last_price  total_area first_day_exposition  rooms  \
5             10   2890000.0       30.40  2018-09-10T00:00:00      1   
8             20   2900000.0       33.16  2018-05-23T00:00:00      1   
12            10   3890000.0       54.00  2016-06-30T00:00:00      2   
22            20   5000000.0       58.00  2017-04-24T00:00:00      2   
30            12   2200000.0       32.80  2018-02-19T00:00:00      1   

    ceiling_height  floors_total  living_area  floor is_apartment  studio  \
5              NaN          12.0        14.40      5          NaN   False   
8              NaN          27.0        15.43     26          NaN   False   
12             NaN           5.0        30.00      5          NaN   False   
22            2.75          25.0        30.00     15          NaN   False   
30             NaN           9.0          NaN      2          NaN   False   

    open_plan  
5       False  
8       False  
12      False  
22      False  
30      False

open_plan  kitchen_area  balcony              locality_name  \
5       False          9.10      NaN  городской посёлок Янино-1   
8       False          8.81      NaN             посёлок Мурино   
12      False          9.00      0.0                  Сертолово   
22      False         11.00      2.0            деревня Кудрово   
30      False           NaN      NaN                   Коммунар   

    airports_nearest  cityCenters_nearest  parks_around3000  parks_nearest  \
5                NaN                  NaN               NaN            NaN   
8                NaN                  NaN               NaN            NaN   
12               NaN                  NaN               NaN            NaN   
22               NaN                  NaN               NaN            NaN   
30               NaN                  NaN               NaN            NaN   

    ponds_around3000  ponds_nearest  days_exposition  
5                NaN            NaN             55.0  
8                NaN            NaN            189.0  
12               NaN            NaN             90.0  
22               NaN            NaN             60.0  
30               NaN            NaN             63.0


 Строки датафрейма, где у столбца 'parks_nearest' обнаружены пропущенные значения:


total_images  last_price  total_area first_day_exposition  rooms  \
1             7   3350000.0       40.40  2018-12-04T00:00:00      1   
5            10   2890000.0       30.40  2018-09-10T00:00:00      1   
6             6   3700000.0       37.30  2017-11-02T00:00:00      1   
7             5   7915000.0       71.60  2019-04-18T00:00:00      2   
8            20   2900000.0       33.16  2018-05-23T00:00:00      1   

   ceiling_height  floors_total  living_area  floor is_apartment  studio  \
1             NaN          11.0        18.60      1          NaN   False   
5             NaN          12.0        14.40      5          NaN   False   
6             NaN          26.0        10.60      6          NaN   False   
7             NaN          24.0          NaN     22          NaN   False   
8             NaN          27.0        15.43     26          NaN   False   

   open_plan  
1      False  
5      False  
6      False  
7      False  
8      False

open_plan  kitchen_area  balcony              locality_name  \
1      False         11.00      2.0             посёлок Шушары   
5      False          9.10      NaN  городской посёлок Янино-1   
6      False         14.40      1.0          посёлок Парголово   
7      False         18.90      2.0            Санкт-Петербург   
8      False          8.81      NaN             посёлок Мурино   

   airports_nearest  cityCenters_nearest  parks_around3000  parks_nearest  \
1           12817.0              18603.0               0.0            NaN   
5               NaN                  NaN               NaN            NaN   
6           52996.0              19143.0               0.0            NaN   
7           23982.0              11634.0               0.0            NaN   
8               NaN                  NaN               NaN            NaN   

   ponds_around3000  ponds_nearest  days_exposition  
1               0.0            NaN             81.0  
5               NaN            NaN             55.0  
6               0.0            NaN            155.0  
7               0.0            NaN              NaN  
8               NaN            NaN            189.0


 Строки датафрейма, где у столбца 'ponds_around3000' обнаружены пропущенные значения:


total_images  last_price  total_area first_day_exposition  rooms  \
5             10   2890000.0       30.40  2018-09-10T00:00:00      1   
8             20   2900000.0       33.16  2018-05-23T00:00:00      1   
12            10   3890000.0       54.00  2016-06-30T00:00:00      2   
22            20   5000000.0       58.00  2017-04-24T00:00:00      2   
30            12   2200000.0       32.80  2018-02-19T00:00:00      1   

    ceiling_height  floors_total  living_area  floor is_apartment  studio  \
5              NaN          12.0        14.40      5          NaN   False   
8              NaN          27.0        15.43     26          NaN   False   
12             NaN           5.0        30.00      5          NaN   False   
22            2.75          25.0        30.00     15          NaN   False   
30             NaN           9.0          NaN      2          NaN   False   

    open_plan  
5       False  
8       False  
12      False  
22      False  
30      False

open_plan  kitchen_area  balcony              locality_name  \
5       False          9.10      NaN  городской посёлок Янино-1   
8       False          8.81      NaN             посёлок Мурино   
12      False          9.00      0.0                  Сертолово   
22      False         11.00      2.0            деревня Кудрово   
30      False           NaN      NaN                   Коммунар   

    airports_nearest  cityCenters_nearest  parks_around3000  parks_nearest  \
5                NaN                  NaN               NaN            NaN   
8                NaN                  NaN               NaN            NaN   
12               NaN                  NaN               NaN            NaN   
22               NaN                  NaN               NaN            NaN   
30               NaN                  NaN               NaN            NaN   

    ponds_around3000  ponds_nearest  days_exposition  
5                NaN            NaN             55.0  
8                NaN            NaN            189.0  
12               NaN            NaN             90.0  
22               NaN            NaN             60.0  
30               NaN            NaN             63.0


 Строки датафрейма, где у столбца 'ponds_nearest' обнаружены пропущенные значения:


total_images  last_price  total_area first_day_exposition  rooms  \
1             7   3350000.0       40.40  2018-12-04T00:00:00      1   
5            10   2890000.0       30.40  2018-09-10T00:00:00      1   
6             6   3700000.0       37.30  2017-11-02T00:00:00      1   
7             5   7915000.0       71.60  2019-04-18T00:00:00      2   
8            20   2900000.0       33.16  2018-05-23T00:00:00      1   

   ceiling_height  floors_total  living_area  floor is_apartment  studio  \
1             NaN          11.0        18.60      1          NaN   False   
5             NaN          12.0        14.40      5          NaN   False   
6             NaN          26.0        10.60      6          NaN   False   
7             NaN          24.0          NaN     22          NaN   False   
8             NaN          27.0        15.43     26          NaN   False   

   open_plan  
1      False  
5      False  
6      False  
7      False  
8      False

open_plan  kitchen_area  balcony              locality_name  \
1      False         11.00      2.0             посёлок Шушары   
5      False          9.10      NaN  городской посёлок Янино-1   
6      False         14.40      1.0          посёлок Парголово   
7      False         18.90      2.0            Санкт-Петербург   
8      False          8.81      NaN             посёлок Мурино   

   airports_nearest  cityCenters_nearest  parks_around3000  parks_nearest  \
1           12817.0              18603.0               0.0            NaN   
5               NaN                  NaN               NaN            NaN   
6           52996.0              19143.0               0.0            NaN   
7           23982.0              11634.0               0.0            NaN   
8               NaN                  NaN               NaN            NaN   

   ponds_around3000  ponds_nearest  days_exposition  
1               0.0            NaN             81.0  
5               NaN            NaN             55.0  
6               0.0            NaN            155.0  
7               0.0            NaN              NaN  
8               NaN            NaN            189.0


 Строки датафрейма, где у столбца 'days_exposition' обнаружены пропущенные значения:


total_images  last_price  total_area first_day_exposition  rooms  \
0             20  13000000.0       108.0  2019-03-07T00:00:00      3   
7              5   7915000.0        71.6  2019-04-18T00:00:00      2   
44            13   5350000.0        40.0  2018-11-18T00:00:00      1   
45            17   5200000.0        50.6  2018-12-02T00:00:00      2   
46            17   6600000.0        52.1  2019-01-31T00:00:00      2   

    ceiling_height  floors_total  living_area  floor is_apartment  studio  \
0             2.70          16.0         51.0      8          NaN   False   
7              NaN          24.0          NaN     22          NaN   False   
44             NaN          22.0          NaN      3          NaN   False   
45            2.65           9.0         30.3      7          NaN   False   
46            2.60          24.0         29.7      9          NaN   False   

    open_plan  
0       False  
7       False  
44      False  
45      False  
46      False

open_plan  kitchen_area  balcony    locality_name  airports_nearest  \
0       False          25.0      NaN  Санкт-Петербург           18863.0   
7       False          18.9      2.0  Санкт-Петербург           23982.0   
44      False           NaN      1.0  Санкт-Петербург           30471.0   
45      False           7.0      NaN  Санкт-Петербург           30011.0   
46      False           8.3      2.0  Санкт-Петербург           15114.0   

    cityCenters_nearest  parks_around3000  parks_nearest  ponds_around3000  \
0               16028.0               1.0          482.0               2.0   
7               11634.0               0.0            NaN               0.0   
44              11603.0               1.0          620.0               1.0   
45              12872.0               0.0            NaN               0.0   
46              12702.0               0.0            NaN               0.0   

    ponds_nearest  days_exposition  
0           755.0              NaN  
7             NaN              NaN  
44         1152.0              NaN  
45            NaN              NaN  
46            NaN              NaN

<div class="alert alert-info">
<b>Комментарий студента:</b>
<br>
Теперь нужно заполнить пропущенные значения, но, к счастью, среди некоторых данных столбцов есть такие данные, которые можно логически определить чисто из здравого смысла, реже с помощью условий из других значений столбцов.<br>  
<br> Ими относятся: 
    <ul> 'is_apartment' (наличие аппартаментов); </ul>
    <ul> 'balcony' (наличие балконов); </ul>
    <ul> 'locality_name' (название населённого пункта); </ul>
    <ul> 'days_exposition' (длительность размещения объявления (от публикации до снятия)); </ul><br>, а также: 
    <ul> 'floors_total' (количество этажей в доме); </ul> 
    <ul> 'airports_nearest' (расстояние до ближайшего аэропорта в метрах (м)); </ul> 
    <ul> 'parks_nearest' (расстояние до ближайшего парка); </ul>
    <ul> 'ponds_nearest' (расстояние до ближайшего пруда). </ul>          
    
<br>    
    Судя на третьей строчке проектной работы по установлению параметров и проведению предобработки и исследовательского анализа данных с архивными объявлениями о продаже квартир в г. Санкт-Петербург: мы обнаруживаем, что вместо представления данных в столбцах: 'studio' (наличие квартиры-студии) и 'open_plan'. <br>
<br>
     Также данные столбцов 'balcony', 'floors_total', а также 'days_exposition' приведём к типу данных 'int' (насчёт столбца 'days_exposition' - ни один SEO-специалист не будет ослеживать длительность с точностью до минуты, для удобства длительность размещения объявления будет измеряться в целых числах дней).<br>
</div>

In [13]:
data['is_apartment'] = data['is_apartment'].fillna('False')
display(data.head())

total_images  last_price  total_area first_day_exposition  rooms  \
0            20  13000000.0       108.0  2019-03-07T00:00:00      3   
1             7   3350000.0        40.4  2018-12-04T00:00:00      1   
2            10   5196000.0        56.0  2015-08-20T00:00:00      2   
3             0  64900000.0       159.0  2015-07-24T00:00:00      3   
4             2  10000000.0       100.0  2018-06-19T00:00:00      2   

   ceiling_height  floors_total  living_area  floor is_apartment  ...  \
0            2.70          16.0         51.0      8        False  ...   
1             NaN          11.0         18.6      1        False  ...   
2             NaN           5.0         34.3      4        False  ...   
3             NaN          14.0          NaN      9        False  ...   
4            3.03          14.0         32.0     13        False  ...   

   kitchen_area  balcony    locality_name  airports_nearest  \
0          25.0      NaN  Санкт-Петербург           18863.0   
1          11.0      2.0   посёлок Шушары           12817.0   
2           8.3      0.0  Санкт-Петербург           21741.0   
3           NaN      0.0  Санкт-Петербург           28098.0   
4          41.0      NaN  Санкт-Петербург           31856.0   

  cityCenters_nearest  parks_around3000  parks_nearest  ponds_around3000  \
0             16028.0               1.0          482.0               2.0   
1             18603.0               0.0            NaN               0.0   
2             13933.0               1.0           90.0               2.0   
3              6800.0               2.0           84.0               3.0   
4              8098.0               2.0          112.0               1.0   

   ponds_nearest  days_exposition  
0          755.0              NaN  
1            NaN             81.0  
2          574.0            558.0  
3          234.0            424.0  
4           48.0            121.0  

[5 rows x 22 columns]

In [14]:
data['balcony'] = (data['balcony'].fillna(0)).astype('int')
display(data.head())

total_images  last_price  total_area first_day_exposition  rooms  \
0            20  13000000.0       108.0  2019-03-07T00:00:00      3   
1             7   3350000.0        40.4  2018-12-04T00:00:00      1   
2            10   5196000.0        56.0  2015-08-20T00:00:00      2   
3             0  64900000.0       159.0  2015-07-24T00:00:00      3   
4             2  10000000.0       100.0  2018-06-19T00:00:00      2   

   ceiling_height  floors_total  living_area  floor is_apartment  ...  \
0            2.70          16.0         51.0      8        False  ...   
1             NaN          11.0         18.6      1        False  ...   
2             NaN           5.0         34.3      4        False  ...   
3             NaN          14.0          NaN      9        False  ...   
4            3.03          14.0         32.0     13        False  ...   

   kitchen_area  balcony    locality_name  airports_nearest  \
0          25.0        0  Санкт-Петербург           18863.0   
1          11.0        2   посёлок Шушары           12817.0   
2           8.3        0  Санкт-Петербург           21741.0   
3           NaN        0  Санкт-Петербург           28098.0   
4          41.0        0  Санкт-Петербург           31856.0   

  cityCenters_nearest  parks_around3000  parks_nearest  ponds_around3000  \
0             16028.0               1.0          482.0               2.0   
1             18603.0               0.0            NaN               0.0   
2             13933.0               1.0           90.0               2.0   
3              6800.0               2.0           84.0               3.0   
4              8098.0               2.0          112.0               1.0   

   ponds_nearest  days_exposition  
0          755.0              NaN  
1            NaN             81.0  
2          574.0            558.0  
3          234.0            424.0  
4           48.0            121.0  

[5 rows x 22 columns]

In [ ]:
data['locality_name'] = data['locality_name'].fillna('Санкт-Петербург')
#display(data.head())
display(data[1090:1110])

In [15]:
data['locality_name'] = data['locality_name'].fillna('Санкт-Петербург')
display(data.head())

total_images  last_price  total_area first_day_exposition  rooms  \
0            20  13000000.0       108.0  2019-03-07T00:00:00      3   
1             7   3350000.0        40.4  2018-12-04T00:00:00      1   
2            10   5196000.0        56.0  2015-08-20T00:00:00      2   
3             0  64900000.0       159.0  2015-07-24T00:00:00      3   
4             2  10000000.0       100.0  2018-06-19T00:00:00      2   

   ceiling_height  floors_total  living_area  floor is_apartment  ...  \
0            2.70          16.0         51.0      8        False  ...   
1             NaN          11.0         18.6      1        False  ...   
2             NaN           5.0         34.3      4        False  ...   
3             NaN          14.0          NaN      9        False  ...   
4            3.03          14.0         32.0     13        False  ...   

   kitchen_area  balcony    locality_name  airports_nearest  \
0          25.0        0  Санкт-Петербург           18863.0   
1          11.0        2   посёлок Шушары           12817.0   
2           8.3        0  Санкт-Петербург           21741.0   
3           NaN        0  Санкт-Петербург           28098.0   
4          41.0        0  Санкт-Петербург           31856.0   

  cityCenters_nearest  parks_around3000  parks_nearest  ponds_around3000  \
0             16028.0               1.0          482.0               2.0   
1             18603.0               0.0            NaN               0.0   
2             13933.0               1.0           90.0               2.0   
3              6800.0               2.0           84.0               3.0   
4              8098.0               2.0          112.0               1.0   

   ponds_nearest  days_exposition  
0          755.0              NaN  
1            NaN             81.0  
2          574.0            558.0  
3          234.0            424.0  
4           48.0            121.0  

[5 rows x 22 columns]

In [16]:
data_days_exposition_median = data['days_exposition'].median()
data_days_exposition_median

95.0

In [17]:
data['days_exposition'] = (data['days_exposition'].fillna(data_days_exposition_median)).astype('int')
display(data.head())

total_images  last_price  total_area first_day_exposition  rooms  \
0            20  13000000.0       108.0  2019-03-07T00:00:00      3   
1             7   3350000.0        40.4  2018-12-04T00:00:00      1   
2            10   5196000.0        56.0  2015-08-20T00:00:00      2   
3             0  64900000.0       159.0  2015-07-24T00:00:00      3   
4             2  10000000.0       100.0  2018-06-19T00:00:00      2   

   ceiling_height  floors_total  living_area  floor is_apartment  ...  \
0            2.70          16.0         51.0      8        False  ...   
1             NaN          11.0         18.6      1        False  ...   
2             NaN           5.0         34.3      4        False  ...   
3             NaN          14.0          NaN      9        False  ...   
4            3.03          14.0         32.0     13        False  ...   

   kitchen_area  balcony    locality_name  airports_nearest  \
0          25.0        0  Санкт-Петербург           18863.0   
1          11.0        2   посёлок Шушары           12817.0   
2           8.3        0  Санкт-Петербург           21741.0   
3           NaN        0  Санкт-Петербург           28098.0   
4          41.0        0  Санкт-Петербург           31856.0   

  cityCenters_nearest  parks_around3000  parks_nearest  ponds_around3000  \
0             16028.0               1.0          482.0               2.0   
1             18603.0               0.0            NaN               0.0   
2             13933.0               1.0           90.0               2.0   
3              6800.0               2.0           84.0               3.0   
4              8098.0               2.0          112.0               1.0   

   ponds_nearest  days_exposition  
0          755.0               95  
1            NaN               81  
2          574.0              558  
3          234.0              424  
4           48.0              121  

[5 rows x 22 columns]

In [18]:
data_floors_total_median = data['floors_total'].median()
data_floors_total_median

9.0

In [19]:
data['floors_total'] = (data['floors_total'].fillna(data_floors_total_median)).astype('int')
display(data.head(50))

total_images  last_price  total_area first_day_exposition  rooms  \
0             20  13000000.0      108.00  2019-03-07T00:00:00      3   
1              7   3350000.0       40.40  2018-12-04T00:00:00      1   
2             10   5196000.0       56.00  2015-08-20T00:00:00      2   
3              0  64900000.0      159.00  2015-07-24T00:00:00      3   
4              2  10000000.0      100.00  2018-06-19T00:00:00      2   
5             10   2890000.0       30.40  2018-09-10T00:00:00      1   
6              6   3700000.0       37.30  2017-11-02T00:00:00      1   
7              5   7915000.0       71.60  2019-04-18T00:00:00      2   
8             20   2900000.0       33.16  2018-05-23T00:00:00      1   
9             18   5400000.0       61.00  2017-02-26T00:00:00      3   
10             5   5050000.0       39.60  2017-11-16T00:00:00      1   
11             9   3300000.0       44.00  2018-08-27T00:00:00      2   
12            10   3890000.0       54.00  2016-06-30T00:00:00      2   
13            20   3550000.0       42.80  2017-07-01T00:00:00      2   
14             1   4400000.0       36.00  2016-06-23T00:00:00      1   
15            16   4650000.0       39.00  2017-11-18T00:00:00      1   
16            11   6700000.0       82.00  2017-11-23T00:00:00      3   
17             6   4180000.0       36.00  2016-09-09T00:00:00      1   
18             8   3250000.0       31.00  2017-01-27T00:00:00      1   
19            16  14200000.0      121.00  2019-01-09T00:00:00      3   
20            12   6120000.0       80.00  2017-09-28T00:00:00      3   
21            13   3200000.0       31.60  2018-03-14T00:00:00      1   
22            20   5000000.0       58.00  2017-04-24T00:00:00      2   
23            11   2950000.0       32.00  2016-10-29T00:00:00      1   
24             8   6500000.0       97.20  2015-10-31T00:00:00      2   
25             3   6800000.0       76.00  2015-10-01T00:00:00      2   
26             6   4050000.0       60.00  2017-04-28T00:00:00      4   
27            20   7100000.0       70.00  2017-05-12T00:00:00      3   
28             8   4170000.0       44.00  2017-12-13T00:00:00      1   
29             9   8600000.0      100.00  2016-04-09T00:00:00      3   
30            12   2200000.0       32.80  2018-02-19T00:00:00      1   
31             8   7200000.0       67.90  2017-10-26T00:00:00      2   
32             7   4990000.0       60.00  2016-05-22T00:00:00      3   
33             8   4800000.0       73.00  2018-10-15T00:00:00      4   
34             3   3290000.0       33.00  2018-02-04T00:00:00      1   
35             6  15500000.0      149.00  2017-06-26T00:00:00      5   
36            13   3790000.0       45.00  2017-01-25T00:00:00      2   
37            10   1990000.0       45.80  2017-10-28T00:00:00      2   
38            10   3150000.0       40.00  2018-03-29T00:00:00      1   
39            15   5200000.0       54.40  2018-11-29T00:00:00      2   
40             9   3590000.0       36.00  2017-03-15T00:00:00      1   
41            16   7900000.0       74.00  2016-05-04T00:00:00      3   
42            13  22000000.0      161.80  2015-07-08T00:00:00      4   
43            13   9330000.0       48.00  2017-01-10T00:00:00      2   
44            13   5350000.0       40.00  2018-11-18T00:00:00      1   
45            17   5200000.0       50.60  2018-12-02T00:00:00      2   
46            17   6600000.0       52.10  2019-01-31T00:00:00      2   
47            17   3600000.0       56.10  2018-10-18T00:00:00      3   
48            10   3600000.0       33.83  2017-10-03T00:00:00      1   
49             1   3050000.0       30.80  2018-11-22T00:00:00      1   

    ceiling_height  floors_total  living_area  floor is_apartment  ...  \
0             2.70            16        51.00      8        False  ...   
1              NaN            11        18.60      1        False  ...   
2              NaN             5        34.30      4        False  ...   
3              NaN            14   

In [20]:
#data.query("locality_name != 'Санкт-Петербург' ")
#data.query('airports_nearest != airports_nearest')
#data[data['airports_nearest'].isna()]

data_unique = data['locality_name'].unique()
data_unique

array(['Санкт-Петербург', 'посёлок Шушары', 'городской посёлок Янино-1',
       'посёлок Парголово', 'посёлок Мурино', 'Ломоносов', 'Сертолово',
       'Петергоф', 'Пушкин', 'деревня Кудрово', 'Коммунар', 'Колпино',
       'поселок городского типа Красный Бор', 'Гатчина', 'поселок Мурино',
       'деревня Фёдоровское', 'Выборг', 'Кронштадт', 'Кировск',
       'деревня Новое Девяткино', 'посёлок Металлострой',
       'посёлок городского типа Лебяжье',
       'посёлок городского типа Сиверский', 'поселок Молодцово',
       'поселок городского типа Кузьмоловский',
       'садовое товарищество Новая Ропша', 'Павловск',
       'деревня Пикколово', 'Всеволожск', 'Волхов', 'Кингисепп',
       'Приозерск', 'Сестрорецк', 'деревня Куттузи', 'посёлок Аннино',
       'поселок городского типа Ефимовский', 'посёлок Плодовое',
       'деревня Заклинье', 'поселок Торковичи', 'поселок Первомайское',
       'Красное Село', 'посёлок Понтонный', 'Сясьстрой', 'деревня Старая',
       'деревня Лесколово', '

In [21]:
#data.query("locality_name != 'Санкт-Петербург' ")
#data_unique.to_csv('')
#data['locality_name'].isin('Кузьмоловский')

In [22]:
#if(data[data['parks_around3000'] == 0.0]):

#data_mean.loc[data_mean['total_income'] != data_mean['total_income'], 'total_income'] = data_mean_total_income_mean  
#display(data_mean.head(30))

  
#data['parks_around3000'] = (data['parks_around3000']).astype('int')
#if()
#data['parks_around3000'] = data[data['parks_around3000'] == 0.0]

#data.loc[:, 'parks_around3000'] = (data.loc[:, 'parks_around3000']).astype('int')
 

In [23]:
data.loc[~(data['parks_around3000'] != data['parks_around3000']), 'parks_nearest'] = (data.loc[:, 'parks_nearest'].fillna(0)).astype('int')

display(data.head(50)) 

total_images  last_price  total_area first_day_exposition  rooms  \
0             20  13000000.0      108.00  2019-03-07T00:00:00      3   
1              7   3350000.0       40.40  2018-12-04T00:00:00      1   
2             10   5196000.0       56.00  2015-08-20T00:00:00      2   
3              0  64900000.0      159.00  2015-07-24T00:00:00      3   
4              2  10000000.0      100.00  2018-06-19T00:00:00      2   
5             10   2890000.0       30.40  2018-09-10T00:00:00      1   
6              6   3700000.0       37.30  2017-11-02T00:00:00      1   
7              5   7915000.0       71.60  2019-04-18T00:00:00      2   
8             20   2900000.0       33.16  2018-05-23T00:00:00      1   
9             18   5400000.0       61.00  2017-02-26T00:00:00      3   
10             5   5050000.0       39.60  2017-11-16T00:00:00      1   
11             9   3300000.0       44.00  2018-08-27T00:00:00      2   
12            10   3890000.0       54.00  2016-06-30T00:00:00      2   
13            20   3550000.0       42.80  2017-07-01T00:00:00      2   
14             1   4400000.0       36.00  2016-06-23T00:00:00      1   
15            16   4650000.0       39.00  2017-11-18T00:00:00      1   
16            11   6700000.0       82.00  2017-11-23T00:00:00      3   
17             6   4180000.0       36.00  2016-09-09T00:00:00      1   
18             8   3250000.0       31.00  2017-01-27T00:00:00      1   
19            16  14200000.0      121.00  2019-01-09T00:00:00      3   
20            12   6120000.0       80.00  2017-09-28T00:00:00      3   
21            13   3200000.0       31.60  2018-03-14T00:00:00      1   
22            20   5000000.0       58.00  2017-04-24T00:00:00      2   
23            11   2950000.0       32.00  2016-10-29T00:00:00      1   
24             8   6500000.0       97.20  2015-10-31T00:00:00      2   
25             3   6800000.0       76.00  2015-10-01T00:00:00      2   
26             6   4050000.0       60.00  2017-04-28T00:00:00      4   
27            20   7100000.0       70.00  2017-05-12T00:00:00      3   
28             8   4170000.0       44.00  2017-12-13T00:00:00      1   
29             9   8600000.0      100.00  2016-04-09T00:00:00      3   
30            12   2200000.0       32.80  2018-02-19T00:00:00      1   
31             8   7200000.0       67.90  2017-10-26T00:00:00      2   
32             7   4990000.0       60.00  2016-05-22T00:00:00      3   
33             8   4800000.0       73.00  2018-10-15T00:00:00      4   
34             3   3290000.0       33.00  2018-02-04T00:00:00      1   
35             6  15500000.0      149.00  2017-06-26T00:00:00      5   
36            13   3790000.0       45.00  2017-01-25T00:00:00      2   
37            10   1990000.0       45.80  2017-10-28T00:00:00      2   
38            10   3150000.0       40.00  2018-03-29T00:00:00      1   
39            15   5200000.0       54.40  2018-11-29T00:00:00      2   
40             9   3590000.0       36.00  2017-03-15T00:00:00      1   
41            16   7900000.0       74.00  2016-05-04T00:00:00      3   
42            13  22000000.0      161.80  2015-07-08T00:00:00      4   
43            13   9330000.0       48.00  2017-01-10T00:00:00      2   
44            13   5350000.0       40.00  2018-11-18T00:00:00      1   
45            17   5200000.0       50.60  2018-12-02T00:00:00      2   
46            17   6600000.0       52.10  2019-01-31T00:00:00      2   
47            17   3600000.0       56.10  2018-10-18T00:00:00      3   
48            10   3600000.0       33.83  2017-10-03T00:00:00      1   
49             1   3050000.0       30.80  2018-11-22T00:00:00      1   

    ceiling_height  floors_total  living_area  floor is_apartment  ...  \
0             2.70            16        51.00      8        False  ...   
1              NaN            11        18.60      1        False  ...   
2              NaN             5        34.30      4        False  ...   
3              NaN            14   

In [24]:
#(data.query("ponds_around3000 == 0 ")).astype('int')
#data['ponds_around3000'] = (data['ponds_around3000']).astype('int')
#data.loc[:, 'ponds_around3000'] = (data.loc[:, 'ponds_around3000']).astype('int')

In [25]:
data.loc[~(data['ponds_around3000'] != data['ponds_around3000']), 'ponds_nearest'] = (data.loc[:, 'ponds_nearest'].fillna(0.0)).astype('int')

display(data.head(50))  

total_images  last_price  total_area first_day_exposition  rooms  \
0             20  13000000.0      108.00  2019-03-07T00:00:00      3   
1              7   3350000.0       40.40  2018-12-04T00:00:00      1   
2             10   5196000.0       56.00  2015-08-20T00:00:00      2   
3              0  64900000.0      159.00  2015-07-24T00:00:00      3   
4              2  10000000.0      100.00  2018-06-19T00:00:00      2   
5             10   2890000.0       30.40  2018-09-10T00:00:00      1   
6              6   3700000.0       37.30  2017-11-02T00:00:00      1   
7              5   7915000.0       71.60  2019-04-18T00:00:00      2   
8             20   2900000.0       33.16  2018-05-23T00:00:00      1   
9             18   5400000.0       61.00  2017-02-26T00:00:00      3   
10             5   5050000.0       39.60  2017-11-16T00:00:00      1   
11             9   3300000.0       44.00  2018-08-27T00:00:00      2   
12            10   3890000.0       54.00  2016-06-30T00:00:00      2   
13            20   3550000.0       42.80  2017-07-01T00:00:00      2   
14             1   4400000.0       36.00  2016-06-23T00:00:00      1   
15            16   4650000.0       39.00  2017-11-18T00:00:00      1   
16            11   6700000.0       82.00  2017-11-23T00:00:00      3   
17             6   4180000.0       36.00  2016-09-09T00:00:00      1   
18             8   3250000.0       31.00  2017-01-27T00:00:00      1   
19            16  14200000.0      121.00  2019-01-09T00:00:00      3   
20            12   6120000.0       80.00  2017-09-28T00:00:00      3   
21            13   3200000.0       31.60  2018-03-14T00:00:00      1   
22            20   5000000.0       58.00  2017-04-24T00:00:00      2   
23            11   2950000.0       32.00  2016-10-29T00:00:00      1   
24             8   6500000.0       97.20  2015-10-31T00:00:00      2   
25             3   6800000.0       76.00  2015-10-01T00:00:00      2   
26             6   4050000.0       60.00  2017-04-28T00:00:00      4   
27            20   7100000.0       70.00  2017-05-12T00:00:00      3   
28             8   4170000.0       44.00  2017-12-13T00:00:00      1   
29             9   8600000.0      100.00  2016-04-09T00:00:00      3   
30            12   2200000.0       32.80  2018-02-19T00:00:00      1   
31             8   7200000.0       67.90  2017-10-26T00:00:00      2   
32             7   4990000.0       60.00  2016-05-22T00:00:00      3   
33             8   4800000.0       73.00  2018-10-15T00:00:00      4   
34             3   3290000.0       33.00  2018-02-04T00:00:00      1   
35             6  15500000.0      149.00  2017-06-26T00:00:00      5   
36            13   3790000.0       45.00  2017-01-25T00:00:00      2   
37            10   1990000.0       45.80  2017-10-28T00:00:00      2   
38            10   3150000.0       40.00  2018-03-29T00:00:00      1   
39            15   5200000.0       54.40  2018-11-29T00:00:00      2   
40             9   3590000.0       36.00  2017-03-15T00:00:00      1   
41            16   7900000.0       74.00  2016-05-04T00:00:00      3   
42            13  22000000.0      161.80  2015-07-08T00:00:00      4   
43            13   9330000.0       48.00  2017-01-10T00:00:00      2   
44            13   5350000.0       40.00  2018-11-18T00:00:00      1   
45            17   5200000.0       50.60  2018-12-02T00:00:00      2   
46            17   6600000.0       52.10  2019-01-31T00:00:00      2   
47            17   3600000.0       56.10  2018-10-18T00:00:00      3   
48            10   3600000.0       33.83  2017-10-03T00:00:00      1   
49             1   3050000.0       30.80  2018-11-22T00:00:00      1   

    ceiling_height  floors_total  living_area  floor is_apartment  ...  \
0             2.70            16        51.00      8        False  ...   
1              NaN            11        18.60      1        False  ...   
2              NaN             5        34.30      4        False  ...   
3              NaN            14   

In [26]:
#for i in range(1, len(data['locality_name'].unique())):
#    print(data['locality_name'].unique()[i])

for value in data['locality_name'].unique():
    print(value)

Санкт-Петербург
посёлок Шушары
городской посёлок Янино-1
посёлок Парголово
посёлок Мурино
Ломоносов
Сертолово
Петергоф
Пушкин
деревня Кудрово
Коммунар
Колпино
поселок городского типа Красный Бор
Гатчина
поселок Мурино
деревня Фёдоровское
Выборг
Кронштадт
Кировск
деревня Новое Девяткино
посёлок Металлострой
посёлок городского типа Лебяжье
посёлок городского типа Сиверский
поселок Молодцово
поселок городского типа Кузьмоловский
садовое товарищество Новая Ропша
Павловск
деревня Пикколово
Всеволожск
Волхов
Кингисепп
Приозерск
Сестрорецк
деревня Куттузи
посёлок Аннино
поселок городского типа Ефимовский
посёлок Плодовое
деревня Заклинье
поселок Торковичи
поселок Первомайское
Красное Село
посёлок Понтонный
Сясьстрой
деревня Старая
деревня Лесколово
посёлок Новый Свет
Сланцы
село Путилово
Ивангород
Мурино
Шлиссельбург
Никольское
Зеленогорск
Сосновый Бор
поселок Новый Свет
деревня Оржицы
деревня Кальтино
Кудрово
поселок Романовка
посёлок Бугры
поселок Бугры
поселок городского типа Рощино
Кириши

In [27]:
'''cities = []
for value in data['locality_name'].unique():
    if(('городской посёлок' in value) | ('поселок' in value) | ('посёлок городского типа' in value) | 
       ('садовое товарищество' in value) | ('деревня Агалатово' in value) | ('деревня Чудской Бор' in value) | 
       ('посёлок Петро-Славянка' in value) | ('деревня Старое Хинколово' in value) | ('садоводческое некоммерческое товарищество' in value) | 
       ('посёлок Платформа 69-й километр' in value) | ('посёлок Кикерино' in value) | ('садоводческое некоммерческое товарищество Лесная Поляна' in value) |
       ('деревня Хязельки' in value) | ('посёлок Кикерино' in value)):
        continue
    else:
        print(value)
        cities.append(value)'''

"cities = []\nfor value in data['locality_name'].unique():\n    if(('городской посёлок' in value) | ('поселок' in value) | ('посёлок городского типа' in value) | \n       ('садовое товарищество' in value) | ('деревня Агалатово' in value) | ('деревня Чудской Бор' in value) | \n       ('посёлок Петро-Славянка' in value) | ('деревня Старое Хинколово' in value) | ('садоводческое некоммерческое товарищество' in value) | \n       ('посёлок Платформа 69-й километр' in value) | ('посёлок Кикерино' in value) | ('садоводческое некоммерческое товарищество Лесная Поляна' in value) |\n       ('деревня Хязельки' in value) | ('посёлок Кикерино' in value)):\n        continue\n    else:\n        print(value)\n        cities.append(value)"

In [28]:
'''airport_Pulkovo = (app.geocode('аэропорт Пулково').raw['lat'], app.geocode('аэропорт Пулково').raw['lon'])

for value in cities:
    point = (app.geocode(value).raw['lat'], app.geocode(value).raw['lon'])
    print(f'Рассстояние между двумя пунктами аэропорт Пулково и {value}: ', round(gd(airport_Pulkovo, point).m, 1))'''

"airport_Pulkovo = (app.geocode('аэропорт Пулково').raw['lat'], app.geocode('аэропорт Пулково').raw['lon'])\n\nfor value in cities:\n    point = (app.geocode(value).raw['lat'], app.geocode(value).raw['lon'])\n    print(f'Рассстояние между двумя пунктами аэропорт Пулково и {value}: ', round(gd(airport_Pulkovo, point).m, 1))"

In [29]:
cities = []
for value in data['locality_name'].unique():
    print(value)
    cities.append(value)

Санкт-Петербург
посёлок Шушары
городской посёлок Янино-1
посёлок Парголово
посёлок Мурино
Ломоносов
Сертолово
Петергоф
Пушкин
деревня Кудрово
Коммунар
Колпино
поселок городского типа Красный Бор
Гатчина
поселок Мурино
деревня Фёдоровское
Выборг
Кронштадт
Кировск
деревня Новое Девяткино
посёлок Металлострой
посёлок городского типа Лебяжье
посёлок городского типа Сиверский
поселок Молодцово
поселок городского типа Кузьмоловский
садовое товарищество Новая Ропша
Павловск
деревня Пикколово
Всеволожск
Волхов
Кингисепп
Приозерск
Сестрорецк
деревня Куттузи
посёлок Аннино
поселок городского типа Ефимовский
посёлок Плодовое
деревня Заклинье
поселок Торковичи
поселок Первомайское
Красное Село
посёлок Понтонный
Сясьстрой
деревня Старая
деревня Лесколово
посёлок Новый Свет
Сланцы
село Путилово
Ивангород
Мурино
Шлиссельбург
Никольское
Зеленогорск
Сосновый Бор
поселок Новый Свет
деревня Оржицы
деревня Кальтино
Кудрово
поселок Романовка
посёлок Бугры
поселок Бугры
поселок городского типа Рощино
Кириши

In [ ]:
unknown_points = []
app = nm(user_agent="tutorial")
airport_Pulkovo = (app.geocode('аэропорт Пулково').raw['lat'], app.geocode('аэропорт Пулково').raw['lon'])

for value in cities:
    try:
        point = (app.geocode(value).raw['lat'], app.geocode(value).raw['lon'])
        print(f'Расстояние между двумя пунктами: аэропорт Пулково и {value} -', round(gd(airport_Pulkovo, point).m, 1), 'метров')
    except:
        print(f'Населённый пункт {value} не найден!')
        unknown_points.append(value)

Расстояние между двумя пунктами: аэропорт Пулково и Санкт-Петербург - 14407.8 метров
Расстояние между двумя пунктами: аэропорт Пулково и посёлок Шушары - 6650.2 метров
Населённый пункт городской посёлок Янино-1 не найден!
Расстояние между двумя пунктами: аэропорт Пулково и посёлок Парголово - 30672.0 метров
Расстояние между двумя пунктами: аэропорт Пулково и посёлок Мурино - 4433098.4 метров
Расстояние между двумя пунктами: аэропорт Пулково и Ломоносов - 30081.0 метров
Расстояние между двумя пунктами: аэропорт Пулково и Сертолово - 38249.8 метров
Расстояние между двумя пунктами: аэропорт Пулково и Петергоф - 21843.3 метров
Расстояние между двумя пунктами: аэропорт Пулково и Пушкин - 12150.6 метров
Расстояние между двумя пунктами: аэропорт Пулково и деревня Кудрово - 20491.0 метров
Расстояние между двумя пунктами: аэропорт Пулково и Коммунар - 21186.8 метров
Расстояние между двумя пунктами: аэропорт Пулково и Колпино - 19833.4 метров
Населённый пункт поселок городского типа Красный Бор 

Расстояние между двумя пунктами: аэропорт Пулково и поселок Заводской - 659857.3 метров
Расстояние между двумя пунктами: аэропорт Пулково и городской посёлок Новоселье - 920401.7 метров
Расстояние между двумя пунктами: аэропорт Пулково и деревня Большие Колпаны - 32790.6 метров
Расстояние между двумя пунктами: аэропорт Пулково и деревня Горбунки - 15785.6 метров
Расстояние между двумя пунктами: аэропорт Пулково и деревня Батово - 57665.8 метров
Расстояние между двумя пунктами: аэропорт Пулково и деревня Заневка - 20491.0 метров
Расстояние между двумя пунктами: аэропорт Пулково и деревня Иссад - 119913.8 метров
Расстояние между двумя пунктами: аэропорт Пулково и Приморск - 111374.5 метров
Расстояние между двумя пунктами: аэропорт Пулково и городской посёлок Фёдоровское - 614993.4 метров
Расстояние между двумя пунктами: аэропорт Пулково и деревня Мистолово - 37278.0 метров
Расстояние между двумя пунктами: аэропорт Пулково и Новая Ладога - 119280.6 метров
Расстояние между двумя пунктами: 

In [ ]:
city_point
#display(data.head(10))
for item in city_point:
    data.loc[(data['airports_nearest'] != data['airports_nearest']) & (data['locality_name'] == item[0]), 'airports_nearest'] = item[1] 
    #data.loc[data['airports_nearest'] != data['airports_nearest'], 'airports_nearest'] = item[1] 
display(data.head(10))

'floors_total' (количество этажей в доме);
'airports_nearest' (расстояние до ближайшего аэропорта в метрах (м));
'parks_nearest' (расстояние до ближайшего парка);
'ponds_nearest'

### 0.2 Шаг 2.2 Приведение столбцов к нужным типам данных.

### 0.3 Шаг 3.1 Подсчёт данных и добавление в новые столбцы.

### 0.4 Шаг 4.1 Проведите исследовательский анализ данных.

## 1 Общий вывод: